### PR TITLE
Move routes config for annotations

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| License       | MIT

Let's make routes configuration when using annotations automatic.

### CODE REVIEW

**Comment by fabpot** [comment]
We cannot merge it now as it fails when the `src/Controller` is empty. But it shouldn't. That's a bug that should be fixed in Symfony first.

**Comment by xabbuh** [comment]
@fabpot Are you sure? This works for me even when the directory is empty. It only fails when the directory does not exist.

File: doctrine/annotations/1.0/config/routes/annotations.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+controllers:
+    resource: ../../src/Controller/
```
Comment (xabbuh) [inline doctrine/annotations/1.0/config/routes/annotations.yaml]: `resource: ../src/Controller/`

Comment (xabbuh) [reply doctrine/annotations/1.0/config/routes/annotations.yaml]: relative paths are interpreted relative to the location of kernel (probably related to symfony/symfony#24982)

**Comment by weaverryan** [comment]
Yes, I believe it only errors out when the directory does NOT exist. But, it should always exist in a new Flex app, thanks to the framework-bundle recipe, correct? I mean, it would be better if it did *not* error if the dir didn't exist, but I think maybe it's not a problem?

File: doctrine/annotations/1.0/config/routes/annotations.yaml (hunk 1 above)
Comment (backbone87) [reply doctrine/annotations/1.0/config/routes/annotations.yaml]: But then `resource: Controller/` should be enough

**Comment by Pierstoval** [comment]
Shouldn't this have been set for `SensioFrameworkBundle` instead of `doctrine/annotations`?

**Comment by weaverryan** [comment]
@Pierstoval I was just thinking the same thing. I'm not sure if it was done on purpose or not. Could you open a new PR to suggest adding it to SensioFrameworkExtraBundle instead?

**Comment by nicolas-grekas** [comment]
The Route annotation doesn't need to be from the bundle: 3.4 has a fine working annotation class
That's why the config is there.
Did you spot any actual issue?

**Comment by backbone87** [comment]
But then this should be in the framework recipe, because If i use doctrine annotations without the framework bundle, this config will cause errors, no?

**Comment by nicolas-grekas** [comment]
framework is a mandatory requirement of flex apps (and it doesn't require annotation, so double "no" :) )

**Comment by fabpot** [comment]
To sum up: it was done on purpose :) Nothing to change here.

File: doctrine/annotations/1.0/config/routes/annotations.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+controllers:
+    resource: ../src/Controller/
```
Comment (backbone87) [inline doctrine/annotations/1.0/config/routes/annotations.yaml]: since this is relative to `%kernel.root_dir%` it should be just `Controller/`

Comment (yceruto) [reply doctrine/annotations/1.0/config/routes/annotations.yaml]: I think it's relative to the entry point (`public/index.php` or `bin/console`) where the kernel is instantiate.

**Pull request merged**